src/renderer/containers/Account/AccountOverview/index.tsx

```diff
@@ -5,56 +5,101 @@ import axios from 'axios';
 
 import DetailPanel from '@renderer/components/DetailPanel';
 import Qr from '@renderer/components/Qr';
-import {getActivePrimaryValidatorConfig} from '@renderer/selectors';
+import {useBooleanState} from '@renderer/hooks';
+import {getActivePrimaryValidatorConfig, getManagedAccounts} from '@renderer/selectors';
 import {formatAddress} from '@renderer/utils/address';
 
 import './AccountOverview.scss';
 
 const AccountOverview: FC = () => {
   const {accountNumber} = useParams();
-  const activePrimaryValidator = useSelector(getActivePrimaryValidatorConfig);
   const [balance, setBalance] = useState<number | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [signingKeyVisible, toggleSigningKeyVisible] = useBooleanState(false);
+  const activePrimaryValidator = useSelector(getActivePrimaryValidatorConfig);
+  const managedAccounts = useSelector(getManagedAccounts);
+  const managedAccount = managedAccounts[accountNumber];
+
+  useEffect(() => {
+    if (signingKeyVisible) toggleSigningKeyVisible();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountNumber]);
 
   useEffect(() => {
     if (!activePrimaryValidator) return;
 
     const fetchData = async (): Promise<void> => {
       const {ip_address: ipAddress, port, protocol} = activePrimaryValidator;
       const address = formatAddress(ipAddress, port, protocol);
+
+      setLoading(true);
       const {data} = await axios.get(`${address}/account_balance/${accountNumber}`);
       setBalance(data.balance);
+      setLoading(false);
     };
 
     fetchData();
   }, [accountNumber, activePrimaryValidator]);
 
+  const getItems = () => {
+    let items = [
+      {
+        key: 'Balance',
+        value: loading ? '-' : balance || '0',
+      },
+      {
+        key: 'Account Number',
+        value: renderAccountNumber(),
+      },
+    ];
+
+    if (managedAccount) {
+      items = [
+        ...items,
+        {
+          key: 'Signing Key',
+          value: renderSigningKey(),
+        },
+      ];
+    }
+
+    return items;
+  };
+
+  const renderSigningKey = () => {
+    if (!managedAccount) return null;
+    return (
+      <>
+        {renderSigningKeyDisplay()}
+        {renderSigningKeyToggle()}
+      </>
+    );
+  };
+
+  const renderSigningKeyDisplay = () => {
+    const {signing_key: signingKey} = managedAccount;
+    return signingKeyVisible ? signingKey : '****************************************************************';
+  };
+
+  const renderSigningKeyToggle = () => {
+    const toggleText = signingKeyVisible ? 'Hide' : 'Show';
+    return (
+      <span className="AccountOverview__signing-key-toggle" onClick={toggleSigningKeyVisible}>
+        {toggleText}
+      </span>
+    );
+  };
+
   const renderAccountNumber = (): ReactNode => (
     <>
-      <div>{accountNumber}</div>
+      <div>{loading ? '-' : accountNumber}</div>
       <Qr className="AccountOverview__qr" text={accountNumber} width={120} />
```

Comment:
Still allowing the QR code to straight up re-render since everything else I tried looks silly.